src/pages/index.js

```diff
@@ -1,84 +1,188 @@
+      });
+  }
+}
+
+const popupWithConfirmation = new ConfirmationOnClick({
+  popupSelector: "#modal-delete-confirmation",
+});
+popupWithConfirmation.setEventListeners();
+
 // popup with form
 const newCardPopup = new PopupWithForm(
   "#profile-add-modal",
-  handleCardFormSubmit
+  handleCardAddFormSubmit
 );
 newCardPopup.setEventListeners();
+
 // edit popup form
-const userInfo = new UserInfo(".profile__title", ".profile__description");
-const popupEditForm = new PopupWithForm("#profile-edit-modal", (formData) => {
-  userInfo.setUserInfo(formData);
-  popupEditForm.close();
-});
+
+const popupEditForm = new PopupWithForm(
+  "#profile-edit-modal",
+  ({ name, about }) => {
+    popupEditForm.saving(true);
+    return api.editProfile({ name, about }).then((editProfile) => {
+      popupEditForm.saving(false);
+      userInformation.setUserInfo(editProfile);
+      popupEditForm.close();
+    });
+  }
+);
 popupEditForm.setEventListeners();
+
 // popup with image
-const popupImage = new PopupWithImage("#preview-image-modal");
-popupImage.setEventListeners();
 
-// section
-function createCard(item) {
-  const cardElement = new Card(item, "#card-template", handleImageClick);
-  return cardElement.getView();
+function handleImageClick(data) {
+  const popupImage = new PopupWithImage("#preview-image-modal");
+  popupImage.setEventListeners();
+  popupImage.open(data);
 }
-
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      const cardElement = createCard(item);
-      section.addItem(cardElement);
-    },
-  },
-  ".cards__list"
-);
-section.renderItems();
-
 // functions
 
-function handleCardFormSubmit(data) {
-  const cardInput = createCard(data);
-  section.addItem(cardInput);
-  newCardPopup.close();
+function createCard(item, _id) {
+  const cardElement = new Card(
+    item,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCardClick,
+    handleImageLike
+  );
+  return cardElement.getView();
 }
 
-function handleImageClick(data) {
-  popupImage.open(data);
+function handleCardAddFormSubmit(data) {
+  newCardPopup.saving(true);
+  api.postNewCard(data).then((card) => {
+    newCardPopup.saving(false);
+    const cardInput = createCard(card);
+    section.addItem(cardInput);
+    newCardPopup.close();
+  });
 }
 
-const editFormValidator = new FormValidation(
-  validationSettings,
-  DOM.profileEditForm
-);
-editFormValidator.enableValidation();
-
-const addFormValidator = new FormValidation(
-  validationSettings,
-  DOM.profileAddForm
-);
-addFormValidator.enableValidation();
+enableValidation(validationSettings);
 
 //event listeners
 DOM.profileEditButton.addEventListener("click", () => {
-  const formData = userInfo.getUserInfo();
+  const formData = userInformation.getUserInfo();
   popupEditForm.setInputValues(formData);
   popupEditForm.open();
 });
 
 // add new card button
 DOM.profileAddButton.addEventListener("click", () => {
-  addFormValidator.toggleButtonState();
```

Comment:
`addFormValidator.toggleButtonState();` is needed here on opening to disable the button